pkg/apis/serving/v1alpha1/revision_validation.go

```diff
@@ -48,10 +49,11 @@ func (r *Revision) checkImmutableFields(ctx context.Context, original *Revision)
 // Validate ensures Revision is properly configured.
 func (r *Revision) Validate(ctx context.Context) *apis.FieldError {
 	errs := serving.ValidateObjectMetadata(r.GetObjectMeta()).ViaField("metadata")
-	errs = errs.Also(r.Spec.Validate(apis.WithinSpec(ctx)).ViaField("spec"))
 	if apis.IsInUpdate(ctx) {
 		old := apis.GetBaseline(ctx).(*Revision)
 		errs = errs.Also(r.checkImmutableFields(ctx, old))
+	} else {
+		errs = errs.Also(r.Spec.Validate(apis.WithinSpec(ctx)).ViaField("spec"))
```

Comment:
Why is this change needed? Do we really not want to validate the spec if we're in an update operation?

Reply:
This is to solve the following scenario: Let's say MaxRevisionTimeout is 600. User creates a revision and sets its timeout to 600. Vendor/Operator reduces MaxRevisionTimeout to 300. If we keep this validation, all existing revisions using > 300 seconds will break. This is to ensure that revisions that passed the validations the first time are honored and keeps working.

checkImmutableFields already ensures that revision spec is not modified, which means that the object still conforms to the original validations.

@mattmoor to correct me if my understanding is wrong

Reply:
I'd rather add a `IsInUpdate` check in the `validateTimoutSeconds`,

Reply:
Think about this some more, it occurs to me that we will default things to the original MaxRevisionTimeout in Revision, but also Configuration and Service.  This check only protects Revision updates, not Configuration or Service updates.

In the situation you describe for e.g. Configuration, our controllers would start failing because we do a full validation even on status updates, and the OOB timeout value would fail validation.

We need to think through all of these contexts a bit better :-/

Reply:
I added logic to `knative/pkg/webhook` so that you can distinguish `/status` updates from other updates.  We should update our validation to be sensitive to this to avoid validation changes leading to spinning controllers.

Reply:
This is now done. Please take a look and let me know if you have any feedback!